### PR TITLE
DAOS-7091 test: Offline reintegration (blank node)

### PR DESCRIPTION
**Pull request by rpadma2**
Test-tag-hw-medium: pr,hw,medium,ib2 offline_reintegration_full

Signed-off-by: rpadma2 <ravindran.padmanabhan@intel.com>

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/util/osa_utils.py [hunk 1]
```diff
@@ -126,6 +127,57 @@ def get_pool_version(self):
...
+        for a particular rank.
+
+        Args:
+            rank (int): daos_engine rank. Defaults to None.
+        Returns:
+            ip_addr (str) : IPAddress for the rank.
+            port_num (str) : Port number for the rank.
+        """
+        ip_addr = [None]
+        port_num = [None]
```
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:142]: This is required if the rank provided is not in the response. The ip_addr and post_num will be returned as None.

[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/util/osa_utils.py:142]: I mean you can return None, None in line 152.

[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:142]: Oh, Got it... Yeah, we can do it... Thanks...

File: src/tests/ftest/util/osa_utils.py [hunk 2]
```diff
@@ -126,6 +127,57 @@ def get_pool_version(self):
         data = self.dmg_command.pool_query(self.pool.uuid)
         return int(data["response"]["version"])
 
+    @fail_on(CommandFailure)
+    def get_ipaddr_for_rank(self, rank=None):
+        """Remove the /mnt/daos/<pool_uid> pool directory
+        for a particular rank.
```
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:133]: Oh, I miss read the line... Sure. Will fix it...

File: src/tests/ftest/util/osa_utils.py [hunk 3]
```diff
@@ -126,6 +127,57 @@ def get_pool_version(self):
...
+        Args:
+            rank (int): daos_engine rank. Defaults to None.
+        Returns:
+            ip_addr (str) : IPAddress for the rank.
+            port_num (str) : Port number for the rank.
+        """
+        ip_addr = [None]
+        port_num = [None]
+        output = self.dmg_command.system_query()
+        members_length = len(output["response"]["members"])
```
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:144]: nice suggestion. will check it...

File: src/tests/ftest/util/osa_utils.py [hunk 4]
```diff
@@ -126,6 +127,57 @@ def get_pool_version(self):
...
+            ip_addr (str): IP address of the daos server.
+                           Defaults to None.
+            port_number (str) : Port number the daos server.
+        """
+        if ip_addr is None or port_num is None:
+            self.log.info("ip_addr : %s port_number: %s", ip_addr, port_num)
+            self.fail("No IP Address or Port number provided")
+        else:
+            expected_ib0_ports = ["31317", "31416"]
+            expected_ib1_ports = ["31417", "31516"]
```
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:168]: hmm.. Let me think about it and see whether we can optimize it...

[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:168]: Another good suggestion. I think I have an idea now. Thanks.

File: src/tests/ftest/util/osa_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:144]: Done.

File: src/tests/ftest/util/osa_utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:168]: Done.

File: src/tests/ftest/util/osa_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:142]: Done.

File: src/tests/ftest/util/osa_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/util/osa_utils.py:133]: Done.

**Review by shimizukko — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by rpadma2** [comment]
This PR is ready for landing. All tests passed.

**Comment by sylviachanoiyee** [comment]
I'm out this out and I think landing is still limited to TB1 and stabilty, please ping gatekeepers for landing when landing to master opens.

**Comment by rpadma2** [comment]
Hi Gatekeeper, Please merge this PR when the master is open. Thanks.

**Comment by rpadma2** [comment]
Hi Gatekeeper, Can we merge this PR? thanks.

**Pull request merged**